e2e-tests/playwright/tests/functional/system_console/sytem_users/column_sort.spec.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {expect} from '@playwright/test';
+
+import {test} from '@e2e-support/test_fixture';
+import {createRandomUser} from '@e2e-support/server';
+
+const EMAIL_COLUMN = 'Email';
+const LAST_LOGIN_COLUMN = 'Last login';
+
+const SIMPLE_EMAIL_REGEX = /\S+@\S+\.\S+/;
+
+test('MM-X Sortable columns should sort the list when clicked', async ({pw, pages}) => {
+    const {adminUser, adminClient} = await pw.initSetup();
+
+    if (!adminUser) {
+        throw new Error('Failed to create admin user');
+    }
+
+    // # Log in as admin
+    const {page} = await pw.testBrowser.login(adminUser);
+
+    for (let i = 0; i < 10; i++) {
+        const randomUser = createRandomUser();
+        await adminClient.createUser(randomUser, '', '');
+    }
+
+    // # Visit system console
+    const systemConsolePage = new pages.SystemConsolePage(page);
+    await systemConsolePage.goto();
+    await systemConsolePage.toBeVisible();
+
+    // # Go to Users section
+    await systemConsolePage.sidebar.goToItem('Users');
+    await systemConsolePage.systemUsers.toBeVisible();
+
+    // * Verify that Email column has aria-sort attribute
+    const userDetailsColumnHeader = await systemConsolePage.systemUsers.getColumnHeader(EMAIL_COLUMN);
+    expect(await userDetailsColumnHeader.isVisible()).toBe(true);
+    expect(userDetailsColumnHeader).toHaveAttribute('aria-sort');
+
+    // # Store the first row's email before sorting
+    const firstRowWithoutSort = await systemConsolePage.systemUsers.getNthRow(1);
+    const firstRowEmailWithoutSort = await firstRowWithoutSort.getByText(SIMPLE_EMAIL_REGEX).allInnerTexts();
+
+    // # Click on the Email column header to sort
+    await systemConsolePage.systemUsers.clickSortOnColumn(EMAIL_COLUMN);
+    await systemConsolePage.systemUsers.isLoadingComplete();
+
+    // # Store the first row's email after sorting
+    const firstRowWithSort = await systemConsolePage.systemUsers.getNthRow(1);
+    const firstRowEmailWithSort = await firstRowWithSort.getByText(SIMPLE_EMAIL_REGEX).allInnerTexts();
+
+    // * Verify that the first row is now different
```

Comment:
Should we expect that it's a certain email since we're generating the users? So we should know which email it should be.

Reply:
Unlike Cypress, our testing setup doesn't start fresh for each test. So, users created in a previous run stick around, making it unclear about the sort order.Our current method is better because it checks if the sort order changed or if the UI responded. The actual correctness of the sort order is tested at the API level, not here. Here, we're just making sure the sort order happens.